### ISSUE TITLE
sign from nearest neighbor attack?

### ISSUE THREAD

**Issue by Xinyu-ustc** [Eric-Wallace/universal-triggers#17]
Hi Wallace, thanks for sharing this work! As I am trying the different token replace strategies in the atttack.py I find a little hard to understand why for increase_loss = False, we apply, e(t+1)=e(t)+g (gradient ascent), while for increase_loss = True, we apply, e(t+1)=e(t)-g. In my understanding, the gradient descent tries to minimize the loss function? And I also try to flipping the sign the original code which gets better attack accuracy compared to the source code? Thanks!

**Comment by Eric-Wallace** [comment]
Hey, I think your right but I need to run the code to verify for sure. The logic in the hotflip code (which is way more tested) is the opposite https://github.com/Eric-Wallace/universal-triggers/blob/ed657674862c965b31e0728d71765d0b6fe18f22/attacks.py#L31 so that logic is probably correct. I will change this line to say `if not increase_loss:` https://github.com/Eric-Wallace/universal-triggers/blob/ed657674862c965b31e0728d71765d0b6fe18f22/attacks.py#L61

**Comment by Eric-Wallace** [comment]
Going to close this issue unless there are further problems.
